src/main/java/pl/allegro/tech/build/axion/release/infrastructure/git/GitRepository.java

```diff
@@ -333,38 +346,18 @@ private TagsOnCommit latestTagsInternal(Pattern pattern, String maybeSinceCommit
 
 
             RevWalk walk = walker(startingCommit);
-            if (!inclusive) {
-                walk.next();
-            }
-
-            Map<String, List<String>> allTags = tagsMatching(pattern, walk);
-
-            RevCommit currentCommit;
-            List<String> currentTagsList;
-            for (currentCommit = walk.next(); currentCommit != null; currentCommit = walk.next()) {
```

Comment:
I am not quite sure if this simplification is not having some serious drawbacks

Reply:
I walker is to get the order of tags, which might not be needed for all cases. I tried to address it with latest.